### PR TITLE
cycling exclusions: supress spurious warning for exclusion points

### LINKED ISSUE #5659: Bad recurrence exclusion warning
See https://cylc.discourse.group/t/cannot-have-more-than-1-repetition-for-zero-duration-recurrence/705/13
```ini
[scheduler]
    cycle point format = %Y
[scheduling]
    initial cycle point = 3000
    [[graph]]
        P1Y ! ^ = "foo[-P1Y] => foo"
[runtime]
    [[foo]]
```

Validation:
```console
$ cylc validate bug
WARNING - Cannot have more than 1 repetition for zero-duration recurrence 3000.
Valid for cylc-8.3.0.dev
```
The warning refers to the exclusion point `^`.

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Closes #5659

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by hjoliver** [comment]
Unfortunately this also surprises valid warnings, e.g. for this:

```
    P1Y ! R3/^/P0Y = foo 
```

**Comment by oliver-sanders** [comment]
Tricky to avoid that, I think either live with the spurious warning or live with the absence of a warning for this case.

**Comment by hjoliver** [comment]
> I think either live with the spurious warning  or live with the absence of a warning for this case

OK, the spurious warning is super confusing, so let's go with the latter option.

File: cylc/flow/cycling/iso8601.py [hunk 1]
```diff
@@ -284,7 +290,20 @@ class ISO8601Sequence(SequenceBase):
 
     """A sequence of ISO8601 date time points separated by an interval.
     Note that an ISO8601Sequence object (may) contain
-    ISO8601ExclusionSequences"""
+    ISO8601ExclusionSequences
+
+    Args:
+        dep_section:
+            The full sequence expresion.
```
Comment (wxtim) [inline cylc/flow/cycling/iso8601.py:297]:
```suggestion
            The full sequence expression.
```

File: cylc/flow/cycling/iso8601.py [hunk 2]
```diff
@@ -269,8 +269,14 @@ def build_exclusions(self, excl_points):
         for point in excl_points:
             try:
                 # Try making an ISO8601Sequence
-                exclusion = ISO8601Sequence(point, self.exclusion_start_point,
-                                            self.exclusion_end_point)
+                exclusion = ISO8601Sequence(
+                    point,
+                    self.exclusion_start_point,
+                    self.exclusion_end_point,
+                    # disable warnings which are logged when exclusion is a
```
Comment (wxtim) [inline cylc/flow/cycling/iso8601.py:276]: Is this comment necessary? Isn't the variable `zero_duration_warning` clear enough?

**Review by wxtim — APPROVED** [review]
Not much to say - looks good.

File: cylc/flow/cycling/iso8601.py (hunk 2 above)
Comment (hjoliver) [reply cylc/flow/cycling/iso8601.py:276]: Fair question, it seems to me that "when exclusion is a time point" is useful context. It's necessarily obvious that a zero duration (from the variable name) is equivalent to a time point.

**Review by hjoliver — APPROVED** [review]

**Comment by hjoliver** [comment]
(Merging; I committed @wxtim 's typo fix)

**Pull request merged**